### DISCUSSION TITLE
Hibernate: Row was updated or deleted by another transaction - related to #27666?

### DISCUSSION THREAD

**Discussion by renegrob** [quarkusio/quarkus#27938] [Q&A]
Hi,

I have an Entity with an element collection:

```java
    @ElementCollection
    @CollectionTable(
            name="SCOPES",
            joinColumns=@JoinColumn(name="ID", referencedColumnName = "ID")
    )
    @Column(name="SCOPE_NAME")
    private Set<String> scopes = new HashSet<>();
```

After upgrading from Quarkus 2.12.1.Final to 2.12.2.Final I get 

```
Caused by: org.hibernate.StaleObjectStateException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect)
```
if I try to to an `entityManager.merge(entity)`. I assume that it is related to #27666 but I'm still trying to figure out how to fix it.
Any help is appreciated!

**Comment by renegrob** [comment]


**Reply by renegrob** [reply]
Looks like it was actually a bugfix! Now, if a merge with the same version is done, it fails.

**Reply by yrodiere** [reply]
Hey. Yes #27666 was a bugfix. Can you please open a bug report **with a reproducer**? Then we can have a look and tell you whether what you're doing was not supposed to work, or whether it's a bug.

**Reply by renegrob** [reply]
@yrodiere  At first I was mislead by the stack trace and thought that it is related to the `@ElementCollection` but it seems that it was not. I don't know if it is worth to do a reproducer since what the unit test did is really not supposed to work (persist and entity with version 0 and then create the another entity object with same id and version 0 and then call `entityManager.merge(entity)` - I have no idea why this worked before but sounds really wrong.)
Sorry for the noise.

**Reply by yrodiere** [reply]
That's... certainly not something I intended to do in #27666, but it may be related to another fix that I'm not aware of.

Anyway, if you're happy, I'm happy. Feel free to open a bug report if this turns out to look like a bug after all :)

[a comment by an automated account is not shown]
